modules/pinger/src/main/java/org/hawkular/component/pinger/MetricPublisher.java

```diff
@@ -47,72 +50,114 @@
 @Stateless
 public class MetricPublisher {
 
-    @javax.annotation.Resource( lookup = "java:/topic/HawkularMetricData")
+    private static void addDataItem(List<Map<String, Object>> mMetrics, String resourceId, long timestamp,
+            Number value, String name) {
+        Map<String, Number> dataMap = new HashMap<>(4);
+        dataMap.put("timestamp", timestamp);
+        dataMap.put("value", value);
+        List<Map<String, Number>> data = new ArrayList<>(1);
+        data.add(dataMap);
+        Map<String, Object> outer = new HashMap<>(4);
+        outer.put("id", resourceId + ".status." + name);
+        outer.put("data", data);
+        mMetrics.add(outer);
+    }
+
+    @javax.annotation.Resource(lookup = "java:/topic/HawkularMetricData")
     javax.jms.Topic topic;
 
-    @javax.annotation.Resource (lookup = "java:/HawkularBusConnectionFactory")
+    @javax.annotation.Resource(lookup = "java:/HawkularBusConnectionFactory")
     ConnectionFactory connectionFactory;
 
     private final PingerConfiguration configuration = PingerConfiguration.defaults();
 
     /**
+     * Serializes the given {@link PingStatus} and then submits it to both {@link #sendToMetricsViaRest(String, List)}
+     * and {@link #publishToTopic(String, List)}.
+     *
+     * @param status
+     *            the {@link PingStatus} to publish
+     */
+    public void publish(PingStatus status) {
+        List<SingleMetric> singleMetrics = new ArrayList<>();
+        List<Map<String, Object>> mMetrics = new ArrayList<>();
+
+        final PingDestination dest = status.getDestination();
+        final String resourceId = dest.getResourceId();
+        final long timestamp = status.getTimestamp();
+        addDataItem(mMetrics, resourceId, timestamp, status.getDuration(), "duration");
+        addDataItem(mMetrics, resourceId, timestamp, status.getCode(), "code");
+
+        SingleMetric singleMetric = new SingleMetric(resourceId + ".status.duration", timestamp,
+                (double) status.getDuration());
+        singleMetrics.add(singleMetric);
+        singleMetric = new SingleMetric(resourceId + ".status.code", timestamp, (double) status.getCode());
+        singleMetrics.add(singleMetric);
+
+        sendToMetricsViaRest(dest.getTenantId(), mMetrics);
+        publishToTopic(dest.getTenantId(), singleMetrics);
+
+    }
+
+    /**
      * Send a list of metric data for a tenant to the Hawkular-metrics service via REST
-     * @param tenantId Name of the tenant
-     * @param metrics List of metrics
+     *
+     * @param tenantId
+     *            Name of the tenant
+     * @param metrics
+     *            List of metrics
      */
     @Asynchronous
-    public void sendToMetricsViaRest(String tenantId, List<Map<String, Object>> metrics) {
+    private void sendToMetricsViaRest(String tenantId, List<Map<String, Object>> metrics) {
```

Comment:
I'm not sure about the effect of `@Asynchronous` on a private method. Does it still work?